layers/core_checks/cc_device.cpp

```diff
@@ -942,3 +956,73 @@ bool CoreChecks::PreCallValidateGetDeviceFaultInfoEXT(VkDevice device, VkDeviceF
     }
     return skip;
 }
+
+bool CoreChecks::PreCallValidateCreatePipelineBinariesKHR(VkDevice device, const VkPipelineBinaryCreateInfoKHR *pCreateInfo,
+                                                          const VkAllocationCallbacks *pAllocator,
+                                                          VkPipelineBinaryHandlesInfoKHR *pBinaries,
+                                                          const ErrorObject &error_obj) const {
+
+    bool skip = false;
+
+    uint32_t pointerCount = 0;
+
+
+    if (pCreateInfo->pipeline != VK_NULL_HANDLE) {
+        auto pipeline = pCreateInfo->pipeline;
+        auto pipeline_state = Get<vvl::Pipeline>(pipeline);
+
+        ASSERT_AND_RETURN_SKIP(pipeline_state);
+
+        if (!(pipeline_state->create_flags & VK_PIPELINE_CREATE_2_CAPTURE_DATA_BIT_KHR)) {
+            skip |= LogError("VUID-VkPipelineBinaryCreateInfoKHR-pipeline-09607", pipeline,
+                             error_obj.location.dot(Field::pCreateInfo).dot(Field::pipeline),
+                             "called on a pipeline created without the "
+                             "VK_PIPELINE_CREATE_2_CAPTURE_DATA_BIT_KHR flag set.");
```

Comment:
I just hit this same error for another extension being worked on, I would add the following as it is easy to go

```
pipeline_info.flags |= VK_PIPELINE_CREATE_2_CAPTURE_DATA_BIT_KHR;
```

and have no compiler error and go "wait, wtf, I AM setting this flag!!!"

```suggestion
                             "VK_PIPELINE_CREATE_2_CAPTURE_DATA_BIT_KHR flag set. (Make sure you set it with VkPipelineCreateFlags2CreateInfoKHR)");
```